zerver/views/presence.py

```diff
@@ -13,25 +13,45 @@
 from zerver.lib.response import json_error, json_success
 from zerver.lib.timestamp import datetime_to_timestamp
 from zerver.lib.validator import check_bool, check_capped_string
-from zerver.models import UserActivity, UserPresence, UserProfile, get_active_user
+from zerver.models import (
+    UserActivity,
+    UserPresence,
+    UserProfile,
+    get_active_user,
+    get_active_user_profile_by_id_in_realm,
+)
 
 
 def get_presence_backend(
-    request: HttpRequest, user_profile: UserProfile, email: str
+    request: HttpRequest, user_profile: UserProfile, user_id_or_email: str
 ) -> HttpResponse:
     # This isn't used by the webapp; it's available for API use by
     # bots and other clients.  We may want to add slim_presence
     # support for it (or just migrate its API wholesale) later.
+    is_email = False
     try:
-        target = get_active_user(email, user_profile.realm)
+        int(user_id_or_email)
+    except ValueError:
+        is_email = True
+
+    try:
+        if is_email:
+            target = get_active_user(user_id_or_email, user_profile.realm)
+        else:
+            target = get_active_user_profile_by_id_in_realm(
+                int(user_id_or_email), user_profile.realm
+            )
```

Comment:
I wish we could make this simpler.  Would it read better to do:
```
try:
     user_id = int(user_id_or_email)
     target = get_active_user_profile_by_id_in_realm(user_id, user_profile.realm)
except ValueError:
     email = user_id_or_email
     target = get_active_user(user_id_or_email, user_profile.realm)
```

maybe all of that inside an outer `try/except` for `UserProfile.DoesNotExist`?

Reply:
Sure I would make the change.